rust/oas-common/src/types/media.rs

```diff
@@ -18,6 +20,14 @@ pub struct Media {
     pub other: serde_json::Map<String, serde_json::Value>,
 }
 
+fn de_from_str<'de, D>(deserializer: D) -> Result<Option<f32>, D::Error>
```

Comment:
I did not test - does this also work for JSON `Number`s or only for strings? I think it should first try to deserialize a `f32` directly and if that fails try to deserialize a str and parse that as `f32` (as here in the function).

Reply:
Could add a test that serializes and deserializes a `Media` to JSON to cover the custom serialization. (Not needed for the PR, there's basically no tests anyway).